# BMS690 - Applied Field Research for Bioinformatics

The primary goal of the course is for students to demonstrate their ability to apply bioinformatics and/or computational biology methods to research or technology development projects. As part of the project, students will conduct a requirements analysis, develop an implementation plan, present status updates, and write a formal final report and give a final presentation.

_Learning Objectives:_
By the end of the course, students will:
- Complete a requirements analysis for a bioinformatics project.
- Wrangle data from public data repositories.
- Research possible bioinformatics and/or computational biology methods.
- Develop an implementation plan for a bioinformatics project.
- Use one or more programming languages (e.g., Python, R) and computing technologies (e.g., Linux cluster, Cloud Computing).
- Develop strategies for testing analysis workflows to ensure that steps are implemented correctly.
- Generate effective data visualizations.

### Required Outcomes:

- Term Project Paper
  - 10-15 pages; 1.5 line spacing with lots of figures
  - Should include Introduction/Background, Description of experiment you are reanalyzing (must include design matrix), Your Analysis Methods, Your Analysis Results, Discussion, References

- Term Project Presentation
  - Pre-recorded presentation (approximately 10 minute long, Roughly 10 slides)
 
- Code Developed for Project
  - Script files, or hosted on Github

### Installation Requirements

Download the most recent versions of R, RStudio, and Seurat (R Toolkit for single-cell genomics) for your laptop:

- [R (version 4.0 or above)](https://www.r-project.org/)
- [RStudio](https://posit.co/download/rstudio-desktop/)
- [Seurat R Package](https://satijalab.org/seurat/articles/install_v5.html)

<br/>

### Bioinformatic Workflow
We will be working through a single-cell RNA sequencing dataset using the R package Seurat. A typical workflow of scRNA-seq data analysis consists of preprocessing, data normalization, dimensionality reduction, clustering, differential gene expression, and gene expression dynamics analysis. 
>See the example scRNA-seq tutorial in Google Colab linked below for more relevant background information on single-cell genomics, to get a better understanding of the wetlab process and techniques for generating single-cell RNA-seq datasets.
<br>

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1ru5q7LtLqsc_Z9bgZiN5sct0n0JzvJrU?usp=sharing)

<img src="https://bioconductor.org/books/3.13/OSCA.intro/images/workflow.png" width="700" height="800">
<i>image source:https://bioconductor.org/books/3.13/OSCA.intro/analysis-overview.html</i>


  
  
